pandas/tests/series/test_constructors.py

```diff
@@ -839,3 +839,27 @@ def test_constructor_cast_object(self):
         s = Series(date_range('1/1/2000', periods=10), dtype=object)
         exp = Series(date_range('1/1/2000', periods=10))
         tm.assert_series_equal(s, exp)
+
+    def test_constructor_generic_timestamp(self):
+        # see gh-15524
+        dtype = np.timedelta64
+        s = Series([], dtype=dtype)
+
+        self.assertTrue(s.empty)
```

Comment:
Actually, as I write this, I see that we do have one of these functions already:
~~~python
assert_equal(a, b, msg="")
~~~